queue_services/entity-filer/tests/unit/filing_processors/test_alteration.py

```diff
@@ -199,3 +202,84 @@ async def test_worker_alteration_court_order(app, session, mocker):
+    alteration_filing['filing']['alteration'] = copy.deepcopy(ALTERATION)
+    alteration_filing['filing']['alteration']['cooperativeAssociationType'] = new_association_type
+    payment_id = str(random.SystemRandom().getrandbits(0x58))
+    filing_submission = create_filing(payment_id, alteration_filing, business_id=business.id)
+
+    filing_meta = FilingMeta()
+
+    # test
+    alteration.process(business=business,
+                       filing_submission=filing_submission,
+                       filing=alteration_filing['filing'],
+                       filing_meta=filing_meta)
+
+    # validate
+    assert business.association_type == new_association_type
+
+
+def test_alteration_coop_rules_and_memorandum(app, session, minio_server):
+    """Assert that the coop association type is altered."""
+    # setup
+    identifier = 'CP1234567'
+    business = create_business(identifier)
+    business.legal_type = Business.LegalTypes.COOP.value
+
+    alteration_filing = copy.deepcopy(FILING_HEADER)
+    alteration_filing['filing']['business']['legalType'] = Business.LegalTypes.COOP.value
+    alteration_filing['filing']['alteration'] = copy.deepcopy(ALTERATION)
+    
+    rules_file_key_uploaded_by_user = upload_file('rules.pdf')
+    memorandum_file_key_uploaded_by_user = upload_file('memorandum.pdf')
+    alteration_filing['filing']['alteration']['rulesFileKey'] = \
+        rules_file_key_uploaded_by_user
+    alteration_filing['filing']['alteration']['rulesFileName'] = 'Rules_File.pdf'
+    alteration_filing['filing']['alteration']['memorandumFileKey'] = \
+        memorandum_file_key_uploaded_by_user
+    alteration_filing['filing']['alteration']['memorandumFileName'] = 'Memorandum_File.pdf'
+    
+    
+    payment_id = str(random.SystemRandom().getrandbits(0x58))
+
+    filing_submission = create_filing(payment_id, alteration_filing, business_id=business.id)
+
+    filing_meta = FilingMeta()
+
+    # test
+    alteration.process(business=business,
+                       filing_submission=filing_submission,
+                       filing=alteration_filing['filing'],
+                       filing_meta=filing_meta)
+
+    business.save()
+
+    documents = business.documents.all()
+
+    for document in documents:
+        if document.filing_id == filing_submission.id:
+            if document.type == DocumentType.COOP_RULES.value:
+                original_rules_key = alteration_filing['filing']['alteration']['rulesFileKey']
+                assert document.file_key == original_rules_key
+                assert MinioService.get_file(document.file_key)
+            elif document.type == DocumentType.COOP_MEMORANDUM.value:
+                original_memorandum_key = \
+                    alteration_filing['filing']['alteration']['memorandumFileKey']
+                assert document.file_key == original_memorandum_key
+                assert MinioService.get_file(document.file_key)
```

Comment:
This would still pass if there were no documents returned for the business.

Reply:
Ahh, I have added more asserts referencing https://github.com/bcgov/lear/blob/main/queue_services/entity-filer/tests/unit/filing_processors/test_incorporation_filing.py#L95